### PR TITLE
Adds two-fer exercise

### PR DESCRIPTION
**Pull request by amscotti**
Hi, this PR is to add a new exercise for `two-fer`, [here is a link to the problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/two-fer)

I'm still learning Dart, so feel free to point out anything that needs to be change or improved.

Thanks! :)

### CODE REVIEW

File: exercises/two-fer/lib/example.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class TwoFer {
+  String twofer(String name) {
+    return "One for ${name is String ? name : 'you'}, one for me.";
```
Comment (jvarness) [inline exercises/two-fer/lib/example.dart]: This exercise could really make good use of default parameters. You might have to add a test where no parameters are passed into the function.

**Comment by jvarness** [comment]
@Stargator @amscotti what do you think about making the module purely functional here with no class body?

**Comment by Stargator** [comment]
@jvarness That seems like a reasonable thing to try. I think we already have at least one other exercise that does that

**Comment by amscotti** [comment]
@jvarness @Stargator Updated, let me know if this was what you were thinking for changes. I kind of feel like the `null` test case should be removed If we leave this as an early exercise. What do you think?

File: exercises/two-fer/lib/example.dart [hunk 2]
```diff
@@ -0,0 +1,3 @@
+String twofer([String name = "you"]) {
+  return "One for ${name is! String ? 'you' : name}, one for me.";
```
Comment (Stargator) [inline exercises/two-fer/lib/example.dart]: The return statement can be replaced with the arrow syntax: ex: `String function(Parameter param) => expression;`

**Comment by Stargator** [comment]
@jvarness @amscotti I would be fine with dropping the null test case as well as making it a core exercise (then the unlocked_by value would be null and the two-fer exercise moved to be after hello-world in config.yaml).

**Comment by amscotti** [comment]
@Stargator Updated, let me know if I missed anything or if you see any other changes. Thanks!

File: exercises/two-fer/.meta/canonical-data.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "exercise": "two-fer",
+  "version": "1.2.0",
+  "cases": [
+    {
```
Comment (Stargator) [inline exercises/two-fer/.meta/canonical-data.json]: Please add the 'no name' test case here so it aligns with the test suite

**Comment by amscotti** [comment]
@Stargator Updated, let me know if that's what you wanted. I left `input` in the JSON but without anything in it.

File: exercises/two-fer/.meta/canonical-data.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      "description": "no name given",
+      "property": "twoFer",
+      "input": {},
```
Comment (Stargator) [inline exercises/two-fer/.meta/canonical-data.json]: I think what we need to do is drop the input and that should be it.

Comment (amscotti) [reply exercises/two-fer/.meta/canonical-data.json]: Removed. :)

**Comment by Stargator** [comment]
@jvarness Is there anything else from your perspective?

**Pull request merged**